sources/packages/api/src/route-controllers/institution-locations/institution-locations.controller.ts

```diff
@@ -231,4 +235,43 @@ export class InstitutionLocationsController extends BaseController {
       locationId,
     );
   }
+
+  /**
+   * Get active application details.
+   * @param applicationId application id.
+   * @returns application Details
+   */
+  @AllowAuthorizedParty(AuthorizedParties.institution)
+  @HasLocationAccess("locationId")
+  @Get(":locationId/active-application/:applicationId")
+  async getActiveApplication(
+    @Param("locationId") locationId: number,
+    @Param("applicationId") applicationId: number,
+  ): Promise<ActiveApplicationDataDto> {
+    const application = await this.applicationService.getActiveApplication(
+      applicationId,
+      locationId,
+    );
+    if (!application) {
+      throw new NotFoundException(
+        `Application id ${applicationId} was not found.`,
+      );
+    }
+    return {
+      applicationStatus: application.applicationStatus,
+      applicationNumber: application.applicationNumber,
+      applicationOfferingIntensity: application.offering?.offeringIntensity,
+      applicationOfferingStartDate: dateString(
+        application.offering?.studyStartDate,
+      ),
+      applicationOfferingEndDate: dateString(
+        application.offering?.studyEndDate,
+      ),
+      applicationLocationName: application.location?.name,
+      applicationStudentName: getUserFullName(application.student?.user),
+      applicationOfferingName: application.offering?.name,
+      applicationProgramDescription: application.pirProgram?.description,
+      applicationProgramName: application.pirProgram?.name,
```

Comment:
Application program name must come from the offering.
The PIR program represents the program selected by the student and could be different from the one indicated by the institution during the PIR process.